third_party/amd/lib/TritonAMDGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -102,10 +102,129 @@ struct LocalLoadOpConversion
   }
 };
 
+struct TransLocalLoadOpConversion
+    : public ConvertOpToLLVMPattern<triton::gpu::LocalLoadOp> {
+public:
+  TransLocalLoadOpConversion(const LLVMTypeConverter &converter,
+                             const AMD::TargetInfo &targetInfo,
+                             PatternBenefit benefit = 2)
+      : ConvertOpToLLVMPattern(converter, benefit), targetInfo(targetInfo) {}
+
+  LogicalResult
+  matchAndRewrite(triton::gpu::LocalLoadOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    MemDescType srcTy = op.getSrc().getType();
+    RankedTensorType dstTy = op.getType();
+    Attribute srcLayout = srcTy.getEncoding();
+    Attribute dstLayout = dstTy.getEncoding();
+
+    if (auto dotLayout = dyn_cast<DotOperandEncodingAttr>(dstLayout)) {
+      if (isa<AMDMfmaEncodingAttr>(dotLayout.getParent())) {
+        if (canUseTransLoad(srcTy, dstTy)) {
+          return lowerSharedToDotOperandTransLL(op, adaptor, getTypeConverter(),
+                                                rewriter);
+        }
+      }
+    }
+    return failure();
+  }
+
+private:
+  bool canUseTransLoad(MemDescType srcTy, RankedTensorType dstTy) const {
+    auto bitwidth = typeConverter->convertType(dstTy.getElementType())
+                        .getIntOrFloatBitWidth();
+
+    auto dotEnc = llvm::cast<DotOperandEncodingAttr>(dstTy.getEncoding());
+    auto mfmaEnc = llvm::dyn_cast<AMDMfmaEncodingAttr>(dotEnc.getParent());
+    // Currently, only transpose loading for 16-bit element types is supported.
+    // Support for 8-bit types will be added in a future patch.
+    if (!mfmaEnc || bitwidth != 16) {
+      return false;
+    }
+
+    // The transposed load lowering logic assumes that double-rate MFMA
```

Comment:
"double-rate MFMA" is not clear. Can you be specific about the mfma instructions you are referring to?